types/parser_driver/value_expr.go

```diff
@@ -90,7 +90,12 @@ func (n *ValueExpr) Restore(ctx *format.RestoreCtx) error {
 		ctx.WritePlain(strconv.FormatFloat(n.GetFloat64(), 'e', -1, 32))
 	case types.KindFloat64:
 		ctx.WritePlain(strconv.FormatFloat(n.GetFloat64(), 'e', -1, 64))
-	case types.KindString, types.KindBytes:
+	case types.KindString:
+		if n.Type.Charset != "" && n.Type.Charset != "utf8mb4" {
+			ctx.WritePlainf("_%s", n.Type.Charset)
```

Comment:
```sql
mysql> select _UTF8'abc', _Utf8'abc', _utf8'abc';
+-----+-----+-----+
| abc | abc | abc |
+-----+-----+-----+
| abc | abc | abc |
+-----+-----+-----+
1 row in set, 3 warnings (0.01 sec)
```

Not sure if we should use `WritePlain` here...

Reply:
i use WriteKeyWord now, PTAL